### PR TITLE
Add description of pvc condition for kubectl describe command

### PR DESCRIPTION
**Pull request by mlmhl**
**What this PR does / why we need it**:

As the title says, add description of `PersistentVolumeClaim.Status.Conditions` for `kubectl describe pvc` command, the output will look like this:

```
Conditions:
  Type       Status  LastProbeTime                     LastTransitionTime                Reason  Message
  ----       ------  -----------------                 ------------------                ------  -------
  Resizing   True    Mon, 01 Jan 0001 00:00:00 +0000   Sat, 10 Feb 2018 19:20:56 +0800           
```

**Release note**:

```release-note
NONE
```

/sig storage
/kind enhancement

### CODE REVIEW

**Comment by krmayankk** [comment]
@mlmhl what was the output before your change ?

**Comment by mlmhl** [comment]
@krmayankk 
The output of `kubectl describe pvc` doesn't show PVC's conditions right now, so this PR only adds the conditions to the output, doesn't modify any other fields.

**Comment by thockin** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/59701)

File: pkg/printers/internalversion/describe.go [hunk 1]
```diff
@@ -1264,6 +1264,20 @@ func describePersistentVolumeClaim(pvc *api.PersistentVolumeClaim, events *api.E
...
 		}
+		if len(pvc.Status.Conditions) > 0 {
+			w.Write(LEVEL_0, "Conditions:\n")
+			w.Write(LEVEL_1, "Type\tStatus\tLastProbeTime\tLastTransitionTime\tReason\tMessage\n")
+			w.Write(LEVEL_1, "----\t------\t-----------------\t------------------\t------\t-------\n")
+			for _, c := range pvc.Status.Conditions {
+				w.Write(LEVEL_1, "%v \t%v \t%s \t%s \t%v \t%v\n",
+					c.Type,
+					c.Status,
+					c.LastProbeTime.Time.Format(time.RFC1123Z),
```
Comment (gnufied) [inline pkg/printers/internalversion/describe.go]: Have you verified the ouput manually? We don't really set `LastProbeTime` currently but a future version of k8s may start setting it, so it makes sense to display it but we don't want to display incorrect value.

Comment (mlmhl) [reply pkg/printers/internalversion/describe.go]: I set `LastProbeTime` to `time.Now()` manually and the output is as below, It looks right:
```
Conditions:
  Type                      Status  LastProbeTime                     LastTransitionTime                Reason  Message
  ----                      ------  -----------------                 ------------------                ------  -------
  FileSystemResizePending   True    Tue, 06 Mar 2018 10:54:09 +0800   Tue, 06 Mar 2018 10:53:59 +0800           Waiting for user to (re-)start a pod to finish file system resize of volume on node.

```

**Comment by mlmhl** [comment]
@adohe  for approve

**Comment by mengqiy** [comment]
/assign @juanvallejo 

@mlmhl please add some test.

**Comment by mlmhl** [comment]
Thanks for reminding @mengqiy, tests are already posted, PTAL.

**Review by juanvallejo — APPROVED** [review]

**Comment by juanvallejo** [comment]
/lgtm

@soltysh fyi

File: pkg/printers/internalversion/describe.go [hunk 2]
```diff
@@ -1264,6 +1264,20 @@ func describePersistentVolumeClaim(pvc *api.PersistentVolumeClaim, events *api.E
 		if pvc.Spec.VolumeMode != nil {
 			w.Write(LEVEL_0, "VolumeMode:\t%v\n", *pvc.Spec.VolumeMode)
 		}
+		if len(pvc.Status.Conditions) > 0 {
```
Comment (soltysh) [inline pkg/printers/internalversion/describe.go]: We are doing something similar in `describeNode`, would you mind creating a helper function and place it there?

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/printers/internalversion/describe.go (hunk 2 above)
Comment (mlmhl) [reply pkg/printers/internalversion/describe.go]: I'm not sure which code we can extract as a helper function, `pvc.Status.Conditions`(its type is `[]PersistentVolumeClaimCondition`) has a different type from `node.Status.Conditions`(its type is `[]NodeCondition`). If we really want to do this, we can only extract the print operation(as follows) as a helper function, Is this what you want?

```go
w.Write(LEVEL_1, "%v \t%v \t%s \t%s \t%v \t%v\n",
	c.Type,
	c.Status,
	c.LastProbeTime.Time.Format(time.RFC1123Z),
	c.LastTransitionTime.Time.Format(time.RFC1123Z),
	c.Reason,
	c.Message)
```

Comment (soltysh) [reply pkg/printers/internalversion/describe.go]: Ah, sorry my bad I didn't notice that.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**